test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -67,17 +68,49 @@ class test_ucp_sockaddr : public ucp_test {
         ASSERT_UCS_OK(status);
     }
 
-    void wait(ucp_worker_h worker, void *request, bool wakeup)
+    static void handle_wakeup(ucp_worker_h send_worker, ucp_worker_h recv_worker)
     {
+        int ret, send_efd, recv_efd;
+        ucs_status_t status;
+
+        ASSERT_UCS_OK(ucp_worker_get_efd(send_worker, &send_efd));
+        ASSERT_UCS_OK(ucp_worker_get_efd(recv_worker, &recv_efd));
+
+        status = ucp_worker_arm(recv_worker);
+        if (status == UCS_ERR_BUSY) {
+            return;
+        }
+        ASSERT_UCS_OK(status);
+
+        status = ucp_worker_arm(send_worker);
+        if (status == UCS_ERR_BUSY) {
+            return;
+        }
+        ASSERT_UCS_OK(status);
+
+        do {
+            struct pollfd pfd[2];
+            pfd[0].fd     = send_efd;
+            pfd[1].fd     = recv_efd;
+            pfd[0].events = POLLIN;
+            pfd[1].events = POLLIN;
+            ret = poll(pfd, 2, -1);
+        } while ((ret < 0) && (errno == EINTR));
+        if (ret < 0) {
+            UCS_TEST_MESSAGE << "poll() failed: " << strerror(errno);
+        }
+
+        EXPECT_GE(ret, 1);
+    }
+
+    void wait(ucp_worker_h send_worker, ucp_worker_h recv_worker, bool wakeup)
```

Comment:
this does not really "wait" for something since it does a single progress and possible goes to sleep with wakeup,
maybe rename it to "progress_with_wakeup" or similar?

Reply:
but it calls a function which would be named "wait_for_wakeup".
also, since this current 'wait' is called not only when wakeup is used, do you think it's ok to use  'wakeup' in the name?